projects/core/src/cms/facade/cms.service.spec.ts

```diff
@@ -69,28 +72,125 @@ describe('CmsService', () => {
     expect(service).toBeTruthy();
   }));
 
-  it('getComponentData should call the store and trigger component load', inject(
-    [CmsService],
-    (service: CmsService) => {
-      spyOn(service, 'getCurrentPage').and.returnValue(of(page));
+  describe('getComponentData', () => {
+    // TODO(issue:6027) - remove this test
+    // TODO:#6027 - should we remove this test?
+    it('getComponentData should call the store and trigger component load', inject(
+      [CmsService],
+      (service: CmsService) => {
+        spyOn(service, 'getCurrentPage').and.returnValue(of(page));
 
-      const testUid = 'test_uid';
-      const mockSelect = createSpy('select').and.returnValue(() => of({}));
-      spyOnProperty(ngrxStore, 'select').and.returnValue(mockSelect);
+        const testUid = 'test_uid';
+        const mockSelect = createSpy('select').and.returnValue(() => of({}));
+        spyOnProperty(ngrxStore, 'select').and.returnValue(mockSelect);
 
-      service
-        .getComponentData(testUid)
-        .pipe(take(1))
-        .subscribe(() => {})
-        .unsubscribe();
+        service
+          .getComponentData(testUid)
+          .pipe(take(1))
+          .subscribe(() => {})
+          .unsubscribe();
 
-      expect(mockSelect).toHaveBeenCalled();
+        expect(mockSelect).toHaveBeenCalled();
 
-      expect(store.dispatch).toHaveBeenCalledWith(
-        new CmsActions.LoadCmsComponent(testUid)
-      );
-    }
-  ));
+        expect(store.dispatch).toHaveBeenCalledWith(
+          new CmsActions.LoadCmsComponent(testUid)
+        );
+      }
+    ));
+
+    describe('when pageContext is NOT provided', () => {
+      it('should use the current page context and dispatch LoadCmsComponent', inject(
+        [CmsService],
+        (service: CmsService) => {
+          const currentPageContext: PageContext = {
+            id: 'current',
+            type: PageType.CONTENT_PAGE,
+          };
+          const nextPageContext: PageContext = {
+            id: 'next',
+            type: PageType.CATALOG_PAGE,
+          };
+
+          spyOn(routingService, 'getPageContext').and.returnValue(
+            of(currentPageContext)
+          );
+          spyOn(routingService, 'getNextPageContext').and.returnValue(
+            of(nextPageContext)
+          );
+
+          const mockLoaderState: LoaderState<boolean> = {
+            success: false,
+            loading: false,
+            error: false,
+          };
+          const mockSelect = createSpy('select').and.returnValue(() =>
+            of(mockLoaderState)
+          );
+          spyOnProperty(ngrxStore, 'select').and.returnValue(mockSelect);
+
+          const uid = 'mockUid';
+          service
+            .getComponentData(uid)
+            .pipe(take(1))
+            .subscribe(() => {})
+            .unsubscribe();
```

Comment:
and `unsubscrbe()` ?

Reply:
Again, copy-paste. As it has `take(1)`, we don't need unsubscribe, but maybe keep it as a good practice, if you don't mind.

Reply:
👍